stage0/src/acpi.rs

```diff
@@ -114,6 +116,118 @@ impl Rsdp {
 
         Ok(())
     }
+
+    pub fn xsdt(&self) -> Result<Option<&Xsdt>, &'static str> {
+        if self.xsdt_address == 0 {
+            Ok(None)
+        } else {
+            Some(Xsdt::new(VirtAddr::new(self.xsdt_address))).transpose()
+        }
+    }
+}
+
+/// Header common for all ACPI tables.
+///
+/// See Section 5.2.6, System Description Table Header, in the ACPI specification for more details.
```

Comment:
imo a link would be nice, esp since it also versions the spec: https://uefi.org/htmlspecs/ACPI_Spec_6_4_html/05_ACPI_Software_Programming_Model/ACPI_Software_Programming_Model.html#system-description-table-header

but we can add them for this and existing structs separately